src/VisualStudio/Core/Def/Implementation/ChangeSignature/IChangeSignatureLanguageService.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.VisualStudio.LanguageServices.Implementation.IntellisenseControls;
+using Microsoft.VisualStudio.Utilities;
+using static Microsoft.VisualStudio.LanguageServices.Implementation.ChangeSignature.ChangeSignatureDialogViewModel;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.ChangeSignature
+{
+    internal interface IChangeSignatureLanguageService : ILanguageService
+    {
+        Task<IntellisenseTextBoxViewModel[]> CreateViewModelsAsync(
+            string[] rolesCollectionType,
+            string[] rolesCollectionName,
+            int insertPosition,
+            Document document,
+            string documentText,
+            IContentType contentType,
+            IntellisenseTextBoxViewModelFactory intellisenseTextBoxViewModelFactory,
+            CancellationToken cancellationToken);
+
+        void GeneratePreviewGrammar(AddedParameterViewModel addedParameterViewModel, List<SymbolDisplayPart> displayParts);
```

Comment:
```suggestion
        void GeneratePreviewDisplayParts(AddedParameterViewModel addedParameterViewModel, List<SymbolDisplayPart> displayParts);
```

I admit "Grammar" wasn't obvious to me what this was. See comment elsewhere too that this should probably return an `ImmutableArray<SymbolDisplayPart>` rather than mutating the one going in to simplify the understanding of what is flowing where.